src/glados/core.py

```diff
@@ -148,4 +197,24 @@ def request(self, request: GladosRequest):
         -------
 
         """
-        return self.router.exec_route(request)
+        # DataStore actions if enabled
+        if self.has_datastore():
+            request.set_datastore(self.datastore)
+            request.set_interaction_from_datastore()
+
+        response = self.router.exec_route(request)
+
+        if self.has_datastore() and request.auto_link and request.new_interaction:
+            try:
+                request.link_interaction_to_message_response(
+                    request.new_interaction, response
+                )
+            except Exception as e:
+                logging.error(
+                    f"error linking response to interaction: {e} response: {response}"
+                )
+
+        if self.has_datastore():
```

Comment:
This should likely be in a `try/except/finally` block, so that if there is an exception raised in `exec_route` the session is still closed (probably by having the `session` be rolled back).

Reply:
I am expecting that some subsets of plugins will return improper responses for this and this was put in before the auto_link option. With the autolink option we just need to make sure that it gets set as needed.. Maybe make gen_new_interaction enable the auto linking